docs/public/specifications/tests/Connectivity-TP/connectivity04.md

```diff
@@ -0,0 +1,254 @@
+   using the *WHOIS String* ("WHOIS Query"). E.g., on Linux:
+
+```
+whois -h riswhois.ripe.net " -F -M 192.0.2.10"
+```
+
+3. [Send] *WHOIS Query* to the *RIS Whois Server*.
+
+4. If there is no response, output *[CN04_ERROR_PREFIX_DATABASE]* and exit this lookup.
+
+5. Extract the string (non-empty line not prepended with "%") from the response, and do:
+   1. If there is no such string, output *[CN04_EMPTY_PREFIX_SET]* and exit this lookup.
+   2. Extract the IP prefix from the second field of the string.
+   3. If it was not possible to extract the IP prefix (i.e., malformed response),
+      output *[CN04_ERROR_PREFIX_DATABASE]* and exit this lookup.
+
+6. Return the IP prefix.
+
+## Intercase dependencies
+
+None
+
+## Terminology
+
+* "Concatenate" - The term is used to refer to the conversion of a TXT
+  resource record’s data to a single contiguous string, as specified in [RFC
+  7208, section 3.3][RFC7208#3.3].
+
+* "Send" - The term is used when a query is sent to
+  a specific server (server IP address).
+
+[Argument List]:                                                https://github.com/zonemaster/zonemaster-engine/blob/master/docs/logentry_args.md
+[CN04_EMPTY_PREFIX_SET]:                                        #outcomes
+[CN04_ERROR_PREFIX_DATABASE]:                                   #outcomes
+[CN04_IPV4_DIFFERENT_PREFIX]:                                   #outcomes
+[CN04_IPV4_SAME_PREFIX]:                                        #outcomes
+[CN04_IPV6_DIFFERENT_PREFIX]:                                   #outcomes
+[CN04_IPV6_SAME_PREFIX]:                                        #outcomes
+[Concatenate]:                                                  #terminology
+[Connectivity01]:                                               ../Connectivity-TP/connectivity01.md
+[CRITICAL]:                                                     ../SeverityLevelDefinitions.md#critical
+[Cymru Database]:                                               #cymru-prefix-lookup
+[DEBUG]:                                                        ../SeverityLevelDefinitions.md#notice
+[DNS Query and Response Defaults]:                              ../DNSQueryAndResponseDefaults.md
+[DNS Query]:                                                    ../DNSQueryAndResponseDefaults.md#default-setting-in-dns-query
+[DNS Response]:                                                 ./DNSQueryAndResponseDefaults.md#default-handling-of-a-dns-response
+[ERROR]:                                                        ../SeverityLevelDefinitions.md#error
+[INFO]:                                                         ../SeverityLevelDefinitions.md#info
+[IP to ASN Mapping]:                                            https://team-cymru.com/community-services/ip-asn-mapping/#dns
+[Message Tag Specification]:                                    ../../../../internal/templates/specifications/tests/MessageTagSpecification.md
```

Comment:
That link is not used anywhere in the document.

Reply:
For template only.